src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject.cs

```diff
@@ -330,13 +336,13 @@ private static void ComputeFolderNames(uint folderItemID, List<string> names, IV
         }
 
         /// <summary>
-        /// Get the default namespace of the project ("" if not defined, which means global namespace),
+        /// Get the value of "rootnamespace" property of the project ("" if not defined, which means global namespace),
         /// or null if it is unknown or not applicable. 
         /// </summary>
         /// <remarks>
-        /// This only has meaning in C# and is explicitly set to null in VB.
+        /// This property has different meaning between C# and VB, each project type can decide how to interpret the value.
         /// </remarks>>
-        private static string GetDefaultNamespace(IVsHierarchy hierarchy, string language)
+        private static string GetRootNamespacePropertyValue(IVsHierarchy hierarchy)
```

Comment:
Unfortunately, that's not the case.But I'm not too concerned about this scenario, because (1) it's not very common and (2) we are moving legacy to `IWorkspaceProjectContext`